### PR TITLE
datadog-agent: support unprivileged build

### PR DESCRIPTION
**Pull request by maxgio92**
This is a draft to try to support unprivileged build for the Datadog Agent.

### CODE REVIEW

File: datadog-agent/tasks-system_probe.py.patch [hunk 1]
```diff
@@ -0,0 +1,20 @@
+--- a/tasks/system_probe.py
++++ b/tasks/system_probe.py
+@@ -1389,7 +1389,7 @@ def build_object_files(
+
+         java_dir = os.path.join("pkg", "network", "protocols", "tls", "java")
+         ctx.run(f"{sudo} mkdir -p {EMBEDDED_SHARE_JAVA_DIR}")
+-        ctx.run(f"{sudo} install -m644 -oroot -groot {java_dir}/agent-usm.jar {EMBEDDED_SHARE_JAVA_DIR}/agent-usm.jar")
```
Comment (maxgio92) [inline datadog-agent/tasks-system_probe.py.patch:7]: This must be analyzed and tested in depth IMO

**Comment by maxgio92** [comment]
It may be not needed with https://github.com/chainguard-dev/melange/pull/1572

**Comment by smoser** [comment]
I successfully built as un-privileged bwrap runner with this change.  Its very similar to the change you're proposing but does the "patch" at a lower level. with this change, `invoke` thinks it is running as root so it won't bother sudo.



```diff
@@ -67,6 +67,7 @@ environment:
       - python3-dev~3.11 # strictly requires python3.11
       - systemd-dev
       - wget # Required for downloading clang-12 and kernel headers from debian
+      - util-linux-misc # unshare
   environment:
     # CGo allows Go programs to call C code
     CGO_ENABLED: "1"
@@ -143,7 +144,8 @@ pipeline:
 
       # Build once to correctly setup links/generates. The system-probe we end
       # up using will be part of the multicall below.
-      invoke -e system-probe.build \
+      unshare --user --map-root-user \
+        invoke -e system-probe.build \
         --strip-object-files \
         --no-bundle \
         --bundle-ebpf
```

**Comment by maxgio92** [comment]
Love it @smoser, that's elegant and should be transparent, thanks. I guess that no capability is needed as they'll be dropped in the new `invoke -e system-probe.build` process, because it completes succesfully.

**Comment by maxgio92** [comment]
I just applied your suggestion @smoser

**Comment by smoser** [comment]
https://github.com/wolfi-dev/os/pull/33387 i put up. you can ack mine or i can ack yours.

**Comment by maxgio92** [comment]
Superseded by https://github.com/wolfi-dev/os/pull/33387

**Pull request closed without merging**